src/frontend/src/optimizer/plan_visitor/max_one_row_visitor.rs

```diff
@@ -65,8 +65,22 @@ impl PlanVisitor<bool> for MaxOneRowVisitor {
                 eq_set.insert(input_ref.index);
             }
         }
-        eq_set.is_superset(&plan.input().logical_pk().iter().copied().collect())
-            || self.visit(plan.input())
+        let input = plan.input();
+        eq_set.is_superset(&input.logical_pk().iter().copied().collect())
+            || {
+                // We don't have UNIQUE key now. So we hack here to support some complex queries on
+                // system tables.
+                if let Some(scan) = input.as_logical_scan() && scan.table_name() == "pg_namespace" {
```

Comment:
Could please use `PG_NAMESPACE_TABLE_NAME` in `pg_namespace.rs` to replace "pg_namespace" and check the table by `is_sys_table()`?